ext/js/core/fetch-utilities.js

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2024  Yomitan Authors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+import {readResponseJson} from './json.js';
+
+/**
+ * @param {string} url
+ * @returns {Promise<Response>}
+ */
+export async function fetchAsset(url) {
```

Comment:
Don't need to export. Better to be conservative with exports and change it only if there ever becomes a need.